streams/test-utils/src/test/java/org/apache/kafka/streams/MockProcessorContextTest.java

```diff
@@ -0,0 +1,392 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.MockProcessorContext;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.Punctuator;
+import org.apache.kafka.streams.processor.TaskId;
+import org.apache.kafka.streams.processor.To;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.internals.InMemoryKeyValueStore;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class MockProcessorContextTest {
+    /**
+     * Behavioral test demonstrating the use of the context for capturing forwarded values
+     */
+    @Test
+    public void shouldCaptureOutputRecords() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            @Override
+            public void process(final String key, final Long value) {
+                this.context().forward(key + value, key.length() + value);
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+
+        processor.init(context);
+
+        processor.process("foo", 5L);
+        processor.process("barbaz", 50L);
+
+        final Iterator<KeyValue> forwarded = context.forwarded().iterator();
+        assertEquals(forwarded.next(), new KeyValue<>("foo5", 8L));
+        assertEquals(forwarded.next(), new KeyValue<>("barbaz50", 56L));
+        Assert.assertFalse(forwarded.hasNext());
+
+        context.resetForwards();
+
+        assertEquals(context.forwarded().size(), 0);
+    }
+
+    /**
+     * Behavioral test demonstrating the use of the context for capturing forwarded values to specific children
+     */
+    @Test
+    public void shouldCaptureRecordsOutputToChildByName() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            private int count = 0;
+
+            @Override
+            public void process(final String key, final Long value) {
+                final To child = count % 2 == 0 ? To.child("george") : To.child("pete");
+                this.context().forward(key + value, key.length() + value, child);
```

Comment:
maybe add some data that are `forward(key + value)` directly, and verify they are also returned below in `forwarded`?

Reply:
I think you mean adding something like:
```java
this.context().forward(key, value);
```
and verify it gets forwarded to all children?

Reply:
yup.